app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -461,7 +461,9 @@ def save_vm(uuid, attr)
       if attr.has_key?(:volumes_attributes)
         vm.volumes.each do |vm_volume|
           volume_attrs = attr[:volumes_attributes].values.detect {|vol| vol[:id] == vm_volume.id}
-          vm_volume.size_gb = volume_attrs[:size_gb]
+          if @volume_attrs.class == Hash
```

Comment:
This will only cover the case that it is a hash, what about for the other case you mentioned in your commit message ?

Reply:
Good point.  I'll put in a few more tests.  I was curious if we still even need this assignment to size_gb...at least I was not able to find a way to directly test it.  I've actually run this successful in our production environment with the ```vm_volume.size_gb = volume_attrs[:size_gb]``` fully commented out with no bad effect.  But always hard to say where it might be called from.

Reply:
This is triggered when you edit the hardware configuration of a vm. Like cpus, memory for example. In this case changing the size of a volume.

From the work I did in #5569 I can say that we still need this bit of code :)